AddingResponsibilitiesToControllers.md

```diff
@@ -0,0 +1,116 @@
+## Adding new responsibilities to controllers
+
+So often we need to make additions to existing controllers. There can be different approaches to follow while implementing these changes. Approach of implementation to pick is important to keep the code as maintainable as possible, or to avoid breaking other parts.
+
+To make things clearer let's assume we have a screen being used for posting videos.
+
+### Inheritance approach
+
+We can leverage inheritance to initialize a new controller with the new property. We can move common code into a new base view controller and create another controller to be initialized with the asset parameter.
+
+One benefit is that code much more easy to maintain because responsibilities are separated, and we don't think twice while changing something if it affects or breaks another parts.
+
+```swift
+class BaseVideoPostViewController: BaseViewController {
+    private let uploadManager: VideoUploadManager
+        
+    // MARK: - Initialization
+    
+    init(with uploadManager: UploadManager) {
+        self.uploadManager = uploadManager
+        
+        super.init()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - View lifecycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+			...
+    }
+    
+    // MARK: - Actions
+    
+    func postButtonDidTap(_ button: UIButton) {
+			createVideoPost()
+    }
+    
+    // MARK: - Video Upload
+            
+    private func createVideoPost() {
+			...
+    }            
+```
+```swift
+class VideoURLPostViewController: BaseVideoPostViewController {
+    private let videoURL: URL
+
+    init(withVideoURL url: URL, and uploadManager: VideoUploadManager) {
```

Comment:
Swift convention is to not put prefixes on `init` arguments. This should be `init(url: URL, uploadManager: VideoUploadManager)`

Reply:
fixed